src/components/TabSelector/TabSelector.js

```diff
@@ -62,32 +62,43 @@ const getOpacity = (position, routesLength, tabIndex, active) => {
 
         return position.interpolate({
             inputRange,
-            outputRange: _.map(inputRange, (i) => (i === tabIndex ? activeValue : inactiveValue)),
+            outputRange: _.map(inputRange, (i) => (affectedTabs.includes(tabIndex) && i === tabIndex ? activeValue : inactiveValue)),
         });
     }
     return activeValue;
 };
 
-const getBackgroundColor = (position, routesLength, tabIndex) => {
+const getBackgroundColor = (position, routesLength, tabIndex, affectedTabs) => {
     if (routesLength > 1) {
         const inputRange = Array.from({length: routesLength}, (v, i) => i);
 
         return position.interpolate({
             inputRange,
-            outputRange: _.map(inputRange, (i) => (i === tabIndex ? themeColors.border : themeColors.appBG)),
+            outputRange: _.map(inputRange, (i) => (affectedTabs.includes(tabIndex) && i === tabIndex ? themeColors.border : themeColors.appBG)),
         });
     }
     return themeColors.border;
 };
 
 function TabSelector({state, navigation, onTabPress, position}) {
     const {translate} = useLocalize();
+
+    const defaultAffectedAnimatedTabs = Array.from({length: state.routes.length}, (v, i) => i);
+    const [affectedAnimatedTabs, setAffectedAnimatedTabs] = useState(defaultAffectedAnimatedTabs);
+
+    React.useEffect(() => {
+        setTimeout(() => {
+            setAffectedAnimatedTabs(defaultAffectedAnimatedTabs);
+        }, CONST.ANIMATED_TRANSITION);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Can you explain why we add this as we do here? https://github.com/Expensify/App/blob/0cbee2247e48c17eec428b989afead3dc6f196d4/src/components/Form.js#L202

Reply:
I add eslint-disable because putting `defaultAffectedAnimatedTabs` as deps will reset the `affectedAnimatedTabs` in every render, because const defaultAffectedAnimatedTabs will re-define the const in every render and is considered deps change because reference identity is changed.
I think we can remove eslint-disable and use `useMemo`

```javascript
const defaultAffectedAnimatedTabs = useMemo(() => Array.from({length: state.routes.length}, (v, i) => i), [state.routes.length]);
    
 React.useEffect(() => {
    setTimeout(() => {
        setAffectedAnimatedTabs(defaultAffectedAnimatedTabs);
    }, CONST.ANIMATED_TRANSITION);
}, [defaultAffectedAnimatedTabs, state.index]);
```